http/middlewares/oauth.go

```diff
@@ -11,42 +13,91 @@ import (
 	"golang.org/x/oauth2/clientcredentials"
 )
 
-func NewOauthTransport(clientID, clientSecret, tokenURL string, scopes ...string) *oauthConfig {
-	return &oauthConfig{
-		clientID:     clientID,
-		clientSecret: clientSecret,
-		tokenURL:     tokenURL,
-		scopes:       scopes,
-		cache:        cache.New(time.Minute*15, time.Hour),
+func NewOauthTransport(config OauthConfig) *oauthRoundTripper {
+	return &oauthRoundTripper{OauthConfig: config, cache: cache.New(time.Minute*15, time.Hour)}
+}
+
+type OauthConfig struct {
+	ClientID     string
+	ClientSecret string
+	TokenURL     string
+	Scopes       []string
+	Params       map[string]string
+	AuthStyle    string
```

Comment:
```suggestion
	AuthStyle    oauth2.AuthStyle
```

and we get rid of `func (c OauthConfig) getAuthStyle`